python/src/scipp/__init__.py

```diff
@@ -4,6 +4,21 @@
 # @author Simon Heybrock
 
 # flake8: noqa
+import os
+
+if os.name == "nt" and "CONDA_PREFIX" in os.environ:
+    # Due to https://github.com/conda-forge/python-feedstock/issues/444 , combinations of Python3, Anaconda and Windows
+    # don't respect os.add_dll_path(...), which is python's mechanism for setting DLL search directories. Instead we
+    # need to explicitly add it to the PATH environment variable. For non-conda versions of python we want to keep using
+    # the usual python mechanism.
+    #
+    # This is probably due to the following patch in conda-built versions of python:
+    # https://github.com/conda-forge/python-feedstock/blob/289b2a8017ddd000896e525f18867f4caacec6f2/recipe/patches/0020-Add-CondaEcosystemModifyDllSearchPath.patch
+    #
+    import pathlib
+    dll_directory = os.path.abspath(
+        os.path.join(pathlib.Path(__file__).parent.resolve(), "..", "bin"))
```

Comment:
Using `__file__` could cause problems in a non-conda build if the package gets zipped. Not sure if that is possible with scipp, but in any case, better use https://docs.python.org/3/library/importlib.html#importlib.resources.path

Also, it is a bit strange to combine `os.path` and `Path` like this, you could just use `(path_from_resources.parent/'bin').resolve()`